Capstone/All Comments/DetailCommentCell.m

```diff
@@ -0,0 +1,108 @@
+//
+//  DetailCommentCell.m
+//  Capstone
+//
+//  Created by Abel Kelbessa on 8/8/22.
+//
+
+#import "DetailCommentCell.h"
+
+@implementation DetailCommentCell
+
+- (void)awakeFromNib {
+    [super awakeFromNib];
+}
+
+- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
+    [super setSelected:selected animated:animated];
+}
+
+- (void)setComment:(Comment *)comment {
+    self.usernameLabel.text = comment.usernameForDisplaying;
+    self.userCommentLabel.text = comment.comment;
+    self.counterLabel.text = [[NSString alloc] initWithFormat:@"%d", comment.vote];
+    self.userProfileImageView.file = comment.image;
+    [self.userProfileImageView loadInBackground];
+    self.currentUser = [PFUser currentUser];
+    self.commentObjectId = comment.objectId;
+    [self voteStatusChecker];
+    [self getTheCommentFromParse];
+}
+
+- (void)getTheCommentFromParse {
+    PFQuery *query = [PFQuery queryWithClassName:@"Comments"];
+    [query whereKey:@"objectId" equalTo:self.commentObjectId];
+    [query findObjectsInBackgroundWithBlock:^(NSArray *comments, NSError *error) {
+        if (comments != nil) {
+            self.commentFromParse = [comments objectAtIndex:0];
+            self.upvotes = self.commentFromParse[@"upvoted"];
+            self.downvotes = self.commentFromParse[@"downvotedUsers"];
+            [self voteStatusChecker];
+        } else {
+            NSLog(@"%@", error.localizedDescription);
+        }
+    }];
+}
+
+- (IBAction)didTapUpvote:(id)sender {
+    PFQuery *query = [PFQuery queryWithClassName:@"Comments"];
+    [query whereKey:@"objectId" equalTo:self.commentObjectId];
+    [query findObjectsInBackgroundWithBlock:^(NSArray *comments, NSError *error) {
+        if (comments != nil) {
+            self.commentFromParse = [comments objectAtIndex:0];
+            [self.downvoteHeartButton setImage:[UIImage imageNamed:@"arrow.down.heart.png"]forState:UIControlStateNormal];
+            [self.upvoteHeartButton setImage:[UIImage imageNamed:@"arrow.up.heart.fill.png"]forState:UIControlStateNormal];
+            self.upvotes = self.commentFromParse[@"upvoted"];
+            [self.upvotes addObject:self.currentUser.username];
+            self.commentFromParse[@"upvoted"] = [NSArray arrayWithArray:self.upvotes];
```

Comment:
I wonder if, instead of storing `upvotes` and `downvotes`, we could directly update the `commentFromParse` object using a [method found in the NSArray documentation](https://developer.apple.com/documentation/foundation/nsarray?language=objc):

```suggestion
            NSArray<NSString *> *upvotes = self.commentFromParse[@"upvoted"];
            self.commentFromParse[@"upvoted"] = [upvotes arrayByAddingObject:self.currentUser.username];
```

This would reduce the potential for error where the locally stored `upvoted` and `downvoted` arrays gets mismatched with those fields in Parse.